v2/inc/c_logging/logger.h

```diff
@@ -50,10 +56,17 @@ extern "C" {
     void logger_log(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line_no, const char* format, ...);
     void logger_log_with_config(LOGGER_CONFIG logger_config, LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line_no, const char* format, ...);
 
+#define LOGGER_FORMATTING_SYNTAX_CHECK(format, ...) \
+    (void)(0 && printf(format MU_IFCOMMALOGIC(MU_COUNT_ARG(__VA_ARGS__)) __VA_ARGS__));
```

Comment:
>(void) [](http://example.com/codeflow?start=4&length=6)

what happens if you remove this cast?

Reply:
I remember this was like that for a reason. It is carried over from the previous v1 logging.
But I removed it now and we'll see if it comes back. I suspect it might on some gcc flavor - but we'll deal with it then.

Reply:
Yep, as I remembered, it did fail on Linux, so adding it back.